### DISCUSSION TITLE
file picker should ommit 'ignored' files by default

### DISCUSSION THREAD

**Discussion by adsick** [helix-editor/helix#1358] [Ideas] [answered]
just opening a working rust project in Helix meets me with this flood:
![image](https://user-images.githubusercontent.com/24528839/147374579-d8f4f9bd-b857-49b4-9065-b5d81d3ec0c8.png)

programs like [ripgrep](https://github.com/BurntSushi/ripgrep), [fd](https://github.com/sharkdp/fd) respect ```.gitignore``` files and do not display ignored content unless otherwise chosen.

file picker should do that too by default and maybe have some additional options to control this. Also 'folding' could be possible (smth like "56 here ignored files folded here, <Enter> to view them"

**Comment by Omnikar** [comment, marked as answer]
Helix already does this. Does your local copy of the repository have a `.git` folder? It won't work if it doesn't.

**Reply by adsick** [reply]
I'm not quite sure, but all Rust projects have git vcs setup by default, right? the crate on the screenshot is not published but I'll check for .git files

**Reply by adsick** [reply]
Yes, you are right. This particular project was copied from some place and it doesn't have .git files.
BUT file picker should fold/nest directories if they have a lot of files in them as this flat approach won't work in big projects. Particularly it should be the number of lines remaining in the list right now, e.g. we've displayed some top-level files and they took 1/2 of space in our list, next we have a dir which has N files in it. If N is less than the remaining line count in the list then we flatten it, otherwise we fold.
that's not a perfect algorithm but we could iterate from that.
